pydantic_ai_slim/pydantic_ai/mcp.py

```diff
@@ -239,6 +239,13 @@ def _map_tool_result_part(
                 )
             else:
                 assert_never(resource)
+        elif isinstance(part, mcp_types.ResourceLink):
+            return {
```

Comment:
Is the expectation that this resource link is sent straight to the model as JSON, or are we supposed to handle it somehow and pass the actual resource?

If it's supposed to be JSON, we can use `part.model_dump()` here.

Reply:
@medaminezghal I've asked a followup question: https://github.com/modelcontextprotocol/modelcontextprotocol/discussions/872#discussioncomment-13630386

Reply:
@medaminezghal Yeah that's what I'm thinking. I was gonna wait for confirmation but I expect that's the way we'll go, so if you feel like adding it please do!

Reply:
@medaminezghal That'll only work if the resource is on the local filesystem. I think we'll need to use the `resources/read` endpoint: https://modelcontextprotocol.io/specification/2025-06-18/server/resources#reading-resources. In the Python SDK, that's the `await session.read_resource` call from this example: https://github.com/modelcontextprotocol/python-sdk#writing-mcp-clients.